### PR TITLE
make sure to upgrade pip and added some explanations

### PR DESCRIPTION
**Pull request by jspaaks**
**List of related issues or pull requests**

Refs: -

**Describe the changes made in this pull request**

Added instruction to upgrade pip 
Added some explanations of the release steps

**Instructions to review the pull request**

### CODE REVIEW

File: README.dev.rst [hunk 1]
```diff
@@ -109,14 +112,28 @@ In a new terminal, without an activated virtual environment or a venv3 directory
 
 .. code:: shell
 
+    # prepare a new directory
     cd $(mktemp -d --tmpdir howfairis.XXXXXX)
+    
+    # fresh git clone ensures the release has the state of origin/main branch
     git clone https://github.com/fair-software/howfairis.git .
+    
+    # prepare a clean virtual environment
```
Comment (fdiblen) [inline README.dev.rst:121]:
```suggestion
    # prepare a clean virtual environment and activate it
```

File: README.dev.rst [hunk 2]
```diff
@@ -109,14 +112,28 @@ In a new terminal, without an activated virtual environment or a venv3 directory
...
     source venv3/bin/activate
+    
+    # make sure to have a recent version of pip
+    pip install --upgrade pip 
+
+    # install runtime dependencies and publishing dependencies
     pip install --no-cache-dir .
     pip install --no-cache-dir .[publishing]
+    
+    # clean up any existing stuff
```
Comment (fdiblen) [inline README.dev.rst:132]:
```suggestion
    # clean up any previously generated artifacts 
```

**Review by fdiblen — APPROVED** [review]
LVGTM :medal_military:

**Comment by jspaaks** [comment]
Thanks for reviewing!

**Pull request merged**